### DISCUSSION TITLE
predict.rfsrc() Performance

### DISCUSSION THREAD

**Discussion by coltrezi** [kogalur/randomForestSRC#118] [Q&A]
Hello! 

Thank you for your development and maintenance of this library. 

We've been using version 2.9.3 for a while, and we were interested in updating to the latest version, but we've noticed that getting the predict results gets longer with each version. We switched versions and ran the same request to see how the performance has evolved over time. For a prediction request (with 90,000 rows in the input dataframe) the results look like this:

2.9.3: 1.3 minutes
2.10.0: 3.5 minutes
2.11.0: 3.4 minutes
2.12.0: 17 minutes
2.13.0: 20 minutes

We were wondering if this was expected behavior? Let me know if there is any additional information I can provide.

Thanks again! 

Colt

**Comment by kogalur** [comment]
It's a bit hard for us determine the cause of your performance issue(s).  As you can see in the trace below, on a machine with several cores, we process 90,000 rows in 11 seconds on the latest build.  Even in running serially, we see that the user time would be about 140 seconds:

`> x=matrix(rnorm(90000*10),nrow=90000);y=x[,1]+3*x[,2];d=data.frame(y=y,x)`
```
> o=rfsrc(y~.,d)
> o
                         Sample size: 90000
                     Number of trees: 500
           Forest terminal node size: 5
       Average no. of terminal nodes: 10853.44
No. of variables tried at each split: 4
              Total no. of variables: 10
       Resampling used to grow trees: swor
    Resample size used to grow trees: 56880
                            Analysis: RF-R
                              Family: regr
                      Splitting rule: mse *random*
       Number of random split points: 10
                     (OOB) R squared: 0.9983154
                    (OOB) Error rate: 0.016788

> system.time(predict(o))
   user  system elapsed 
 92.518  48.578  11.438 
```

**Reply by coltrezi** [reply]
@kogalur thanks for your response! I am able to run your example and I get a similar performance. I'll try to make a reproducible example and get back.
